### PR TITLE
Use the flux provider bootstrap resources

### PR DESCRIPTION
**Pull request by avnes**
## Breaking change

Manual changes needed:

```bash
cd infrastructure-modules
git checkout $TAG_VERSION

cd sandbox/$REGION>/$CLUSTER_NAME/services

terragrunt state pull > terraform.state # Backup

terragrunt init -upgrade

terragrunt state rm module.platform_fluxcd[0].null_resource.flux_namespace
terragrunt state rm module.platform_fluxcd[0].kubectl_manifest.install
terragrunt state rm module.platform_fluxcd[0].kubectl_manifest.sync
terragrunt state rm module.platform_fluxcd[0].kubernetes_secret.main
terragrunt state rm module.platform_fluxcd[0].data.flux_install.main
terragrunt state rm module.platform_fluxcd[0].data.flux_sync.main
terragrunt state rm module.platform_fluxcd[0].data.kubectl_file_documents.install
terragrunt state rm module.platform_fluxcd[0].data.kubectl_file_documents.sync
terragrunt state rm module.platform_fluxcd[0].github_repository_file.install
terragrunt state rm module.platform_fluxcd[0].github_repository_file.kustomize
terragrunt state rm module.platform_fluxcd[0].github_repository_file.sync

terragrunt state mv module.platform_fluxcd[0].data.github_branch.flux_branch module.platform_fluxcd.data.github_branch.flux_branch
terragrunt state mv module.platform_fluxcd[0].data.github_repository.main module.platform_fluxcd.data.github_repository.main
terragrunt state mv module.platform_fluxcd[0].github_repository_deploy_key.main module.platform_fluxcd.github_repository_deploy_key.main
terragrunt state mv module.platform_fluxcd[0].github_repository_file.flux_monitoring_config_path module.platform_fluxcd.github_repository_file.flux_monitoring_config_path
terragrunt state mv module.platform_fluxcd[0].github_repository_file.platform_apps_init module.platform_fluxcd.github_repository_file.platform_apps_init
terragrunt state mv module.platform_fluxcd[0].tls_private_key.main module.platform_fluxcd.tls_private_key.main

terragrunt import module.platform_fluxcd.flux_bootstrap_git.this flux-system

terragrunt apply
```

### CODE REVIEW

**Comment by avnes** [comment]
Here is proof that the PR works: https://dfds.visualstudio.com/CloudEngineering/_build/results?buildId=571422&view=logs&s=859b8d9a-8fd6-5a5c-6f5e-f84f1990894e

**Comment by avnes** [comment]
https://github.com/fluxcd/terraform-provider-flux/issues/465

File: test/integration/eu-west-1/k8s-qa/services/terragrunt.hcl [hunk 1]
```diff
@@ -2,6 +2,11 @@
 # working directory, into a temporary folder, and execute your Terraform commands in that folder.
 terraform {
   source = "../../../../..//compute/k8s-services"
+
+  before_hook "custom_hook_01" {
+    commands     = ["apply"]
+    execute      = ["${get_parent_terragrunt_dir()}/hooks/custom_hook_01.sh", "${get_parent_terragrunt_dir()}"]
+  }
```
Comment (emil2k) [inline test/integration/eu-west-1/k8s-qa/services/terragrunt.hcl:9]: Nicely done.

**Review by emil2k — APPROVED** [review]
Great job 👍

File: _sub/compute/k8s-fluxcd/main.tf [hunk 2]
```diff
@@ -1,148 +1,69 @@
...
-    "identity.pub" = tls_private_key.main.public_key_pem
-    known_hosts    = local.known_hosts
-  }
+data "github_branch" "flux_branch" {
+  repository = var.repository_name
+  branch     = var.branch
 }
 
 resource "github_repository_deploy_key" "main" {
   title      = "flux-${var.cluster_name}-readonly"
```
[the remark this answers is not in the document]
Comment (avnes) [reply _sub/compute/k8s-fluxcd/main.tf:33]: Yes, It was kept readonly in the name to not cause a destroy/create on the resource

**Pull request merged**